internal/executor/podchecks/pod_checks.go

```diff
@@ -51,3 +60,10 @@ func (pc *PodChecks) GetAction(pod *v1.Pod, podEvents []*v1.Event, timeInState t
 	log.Infof("Pod checks for pod %s returned %s %s\n", pod.Name, resultAction, resultMessage)
 	return resultAction, resultMessage
 }
+
+// If a node is bad, we can have no pod status and no pod events.
+// We should retry the pod rather than wait
+func (pc *PodChecks) isNodeBad(pod *v1.Pod, podEvents []*v1.Event) bool {
```

Comment:
I'm not sure I love the name of this, I'm not sure it'll always be true that the node is bad when it happens.

However I think it does the right thing

Reply:
we could focus on the events and status of the pod, not the node: `hasNoEventsOrStatus()` or something like that

Reply:
Done.